pkg/controller/sync/aggregator/aggregator.go

```diff
@@ -0,0 +1,105 @@
+package aggregator
+
+import (
+	"fmt"
+	gosync "sync"
+
+	"golang.org/x/exp/maps"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type KindName struct {
+	Kind string
+	Name string
+}
+
+func NewGVKAggregator(allowedKeyKinds []string) *GVKAgreggator {
+	return &GVKAgreggator{
+		store:        make(map[KindName]map[schema.GroupVersionKind]struct{}),
+		reverseStore: make(map[schema.GroupVersionKind]map[KindName]struct{}),
+
+		allowedKeyKinds: allowedKeyKinds,
+	}
+}
+
+// todo comments.
+type GVKAgreggator struct {
+	mu gosync.RWMutex
+
+	store        map[KindName]map[schema.GroupVersionKind]struct{}
```

Comment:
nit: Plz add comments to both `store` and `reverseStore` saying their purpose

Reply:
I've added the comments requested here but let's hold off on more until the shape stabilizes here.